tests/components/mqtt/test_init.py

```diff
@@ -64,619 +58,642 @@ def mock_mqtt():
-        assert self.calls[0][0].data["service_data"][mqtt.ATTR_TOPIC] == "test-topic"
-        assert (
-            self.calls[0][0].data["service_data"][mqtt.ATTR_PAYLOAD] == "test-payload"
-        )
+    assert len(calls) == 1
+    assert calls[0][0].data["service_data"][mqtt.ATTR_TOPIC] == "test-topic"
+    assert calls[0][0].data["service_data"][mqtt.ATTR_PAYLOAD] == "test-payload"
 
-    def test_service_call_without_topic_does_not_publish(self):
-        """Test the service call if topic is missing."""
-        self.hass.bus.fire(
-            EVENT_CALL_SERVICE,
-            {ATTR_DOMAIN: mqtt.DOMAIN, ATTR_SERVICE: mqtt.SERVICE_PUBLISH},
-        )
-        self.hass.block_till_done()
-        assert not self.hass.data["mqtt"].async_publish.called
-
-    def test_service_call_with_template_payload_renders_template(self):
-        """Test the service call with rendered template.
-
-        If 'payload_template' is provided and 'payload' is not, then render it.
-        """
-        mqtt.publish_template(self.hass, "test/topic", "{{ 1+1 }}")
-        self.hass.block_till_done()
-        assert self.hass.data["mqtt"].async_publish.called
-        assert self.hass.data["mqtt"].async_publish.call_args[0][1] == "2"
-
-    def test_service_call_with_payload_doesnt_render_template(self):
-        """Test the service call with unrendered template.
-
-        If both 'payload' and 'payload_template' are provided then fail.
-        """
-        payload = "not a template"
-        payload_template = "a template"
-        with pytest.raises(vol.Invalid):
-            self.hass.services.call(
-                mqtt.DOMAIN,
-                mqtt.SERVICE_PUBLISH,
-                {
-                    mqtt.ATTR_TOPIC: "test/topic",
-                    mqtt.ATTR_PAYLOAD: payload,
-                    mqtt.ATTR_PAYLOAD_TEMPLATE: payload_template,
-                },
-                blocking=True,
-            )
-        assert not self.hass.data["mqtt"].async_publish.called
 
-    def test_service_call_with_ascii_qos_retain_flags(self):
-        """Test the service call with args that can be misinterpreted.
+async def test_service_call_without_topic_does_not_publish(hass, mqtt_mock):
+    """Test the service call if topic is missing."""
+    hass.bus.fire(
+        EVENT_CALL_SERVICE,
+        {ATTR_DOMAIN: mqtt.DOMAIN, ATTR_SERVICE: mqtt.SERVICE_PUBLISH},
+    )
+    await hass.async_block_till_done()
+    assert not mqtt_mock.async_publish.called
+
 
-        Empty payload message and ascii formatted qos and retain flags.
-        """
-        self.hass.services.call(
+async def test_service_call_with_template_payload_renders_template(hass, mqtt_mock):
+    """Test the service call with rendered template.
+
+    If 'payload_template' is provided and 'payload' is not, then render it.
+    """
+    mqtt.async_publish_template(hass, "test/topic", "{{ 1+1 }}")
+    await hass.async_block_till_done()
+    assert mqtt_mock.async_publish.called
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5737505)